R/term-weights.R

```
@@ -59,34 +46,23 @@
 #' @examples
 #' terms <- c("pectoral fin", "pelvic fin", "dorsal fin", "paired fin")
 #' IRIs <- sapply(terms, get_term_iri, as = "anatomy")
-#' term_freqs(IRIs)
+#' term_freqs(IRIs, as = "entity")
 #' 
 #' phens <- get_phenotypes(entity = "basihyal bone")
 #' term_freqs(phens$id, as = "phenotype", corpus = "taxon_annotations")
 #' term_freqs(phens$id, as = "phenotype", corpus = "taxa")
 #' 
 #' @export
 term_freqs <- function(x,
-                       as = c("auto", "entity", "quality", "phenotype"),
+                       as = c("entity", "quality", "phenotype"),
                        corpus = c("taxon_annotations", "taxa", "gene_annotations", "genes"),
-                       decodeIRI = TRUE,
                        ...) {
   as <- match.arg(as, several.ok = TRUE)
   corpus <- match.arg(corpus)
 
-  if (as[1] == "auto") {
-    as <- term_category(x)
-    # check for signs of mistakes and auto-correct
-    fracs <- table(as) / length(as)
-    if (any(fracs < 0.1) && any(fracs > 0.9)) {
-      majorType <- names(fracs)[fracs == max(fracs)]
-      as[as != majorType] <- majorType
-    }
-  } else if (length(as) > 1 && length(as) != length(x))
+  if (length(as) > 1 && length(as) != length(x))
     stop("'as' must be a single value, or have the same length as 'x'", call. = FALSE)
-  else if (any(as == "auto"))
-    stop("'auto' can only be applied to all terms", call. = FALSE)
-  
```

I may be missing something but it seems with this and the preceding deletion we are not checking whether someone passes `"auto"` as the value, as was previously allowed. We'd need to catch this and raise `stop()` rather than silently ignoring it.

---

The match.arg function raises an error if you pass in a bogus parameter value for `as`:
```
> phens <- get_phenotypes(entity = "basihyal bone")
> term_freqs(phens$id, as = "auto", corpus = "taxa")
 Error in match.arg(as, several.ok = TRUE) : 
  'arg' should be one of “entity”, “quality”, “phenotype” 
```
The `as` parameter is now defaulted to 
`as = c("entity", "quality", "phenotype"),`
And the `match.arg` function is called like so:
`as <- match.arg(as, several.ok = TRUE)`.
I will admit I found match.arg surprising since it's reading both the default value and the passed in parameter value.

---

I see. 👍🏻